test/unit/org/apache/cassandra/db/lifecycle/LogReplicationSetTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.lifecycle;
+
+import java.io.File;
+import java.util.Vector;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.io.util.FileUtils;
+import org.mockito.Mockito;
+
+public class LogReplicationSetTest
+{
+    private Vector<File> getReplicaFiles(String testName, int nrReplicas)
+    {
+        Vector<File> files = new Vector<>(nrReplicas);
+        for (int i = 0; i < nrReplicas; i++)
+        {
+            files.add(Mockito.spy(createTempFile(testName, i)));
+        }
+        return files;
+    }
+
+    @Test
+    public void testAppendFailedToAll() throws Throwable
+    {
+        int nrReplicas = 2;
+        LogReplicaSet replicas = new LogReplicaSet();
+        Vector<File> spyFiles = getReplicaFiles("testAppendFailedToAll", nrReplicas);
+
+        replicas.addReplicas(spyFiles);
+        spyFiles.forEach(f -> Mockito.when(f.exists()).thenThrow(new RuntimeException()));
+
+        Assert.assertThrows(RuntimeException.class,
+                            () ->
+                            replicas.append(LogRecord.makeAbort(System.currentTimeMillis())));
+    }
+
+    @Test
+    public void testAppendFailedToSome() throws Throwable
+    {
+        int nrReplicas = 2;
+        LogReplicaSet replicas = new LogReplicaSet();
+        Vector<File> spyFiles = getReplicaFiles("testAppendFailedToSome", nrReplicas);
+
+        replicas.addReplicas(spyFiles);
+        Mockito.when(spyFiles.firstElement().exists()).thenThrow(new RuntimeException());
```

Comment:
Is this test case finished?

Reply:
Yes, no error happens. Is there a way to check the log from the test?